### PR TITLE
feat:  Switch to esbuild and ESM first packaging

### PR DESCRIPTION
**Pull request by GordonSmith**
BREAKING CHANGE: Changed type to "module" and dropped UMD support

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith saw a couple of things in the observable-compiler area. I think all the other comments (the earlier ones) were just things I was wondering about.

File: packages/dataflow/package.json [hunk 1]
```diff
@@ -2,50 +2,44 @@
...
-        "compile-umd-watch": "npm run compile-umd -- -w",
-        "bundle": "rollup -c",
-        "bundle-watch": "npm run bundle -- -w",
-        "bundle-test": "rollup -c ./rollup-test.config.mjs",
-        "bundle-test-watch": "npm run bundle-test -- -w",
-        "minimize": "terser dist/index.js -c -m --source-map \"content='dist/index.js.map',url='index.min.js.map'\" -o dist/index.min.js",
-        "gen-legacy-types": "downlevel-dts ./types ./types-3.4",
-        "build": "npm-run-all -p compile-es6 compile-umd -s bundle",
-        "watch": "npm-run-all compile-es6 -p compile-es6-watch bundle-watch",
+        "clean": "rimraf --glob lib* types dist dist-test *.tsbuildinfo .turbo",
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply packages/dataflow/package.json:20]: I think its related to nx + caching (which we are not using) - I had some stray folders from previous testing I suspect.

File: packages/dataflow/spec/index.browser.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000;
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply packages/dataflow/spec/index.browser.ts:1]: For the browser tests, Jasmine will download the Firefox runner and selenium web driver if needed, hence the longer timeout.

File: packages/dataflow/src/activities/concat.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Source, IterableActivity } from "./activity";
+import { Source, IterableActivity } from "./activity.ts";
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply packages/dataflow/src/activities/concat.ts:1]: Its part of the "esm standard".  TypeScript accepts either .js or .ts with a flag.

File: packages/observablehq-compiler/src/writer.ts [hunk 4]
```diff
@@ -43,14 +43,14 @@ export default function define(runtime, observer) {
     }
 
     importDefine(imp: Partial<ParsedImportCell>) {
-        const injections = imp.injections.map(inj => {
+        const injections = imp.injections!.map(inj => {
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply packages/observablehq-compiler/src/writer.ts]: Switched to double check for undefined and then use `[]`

File: packages/observablehq-compiler/src/util.ts [hunk 5]
```diff
@@ -158,7 +158,7 @@ export function omd2notebook(omd: string): ohq.Notebook {
 }
 
 export function fetchEx(url: string, proxyPrefix = "https://api.codetabs.com/v1/proxy/?quest=", proxyPostfix = "") {
-    const matches = url.match(/^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n?]+)/img);
+    const matches = url.match(/^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n?]+)/img) || [url];
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply packages/observablehq-compiler/src/util.ts]: Updated to throw on match error.

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**